lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -648,18 +692,27 @@ export class UserAgentApplication {
         return;
       }
 
-      //if user is not currently logged in and no login_hint is passed
-      if (!userObject && !(extraQueryParameters && (extraQueryParameters.indexOf(Constants.login_hint) !== -1))) {
+      // If no session exists, prompt the user to login.
+      if (!userObject && !!(request.sid  || request.loginHint)) {
         this.logger.info("User login is required");
         reject(ErrorCodes.userLoginError + Constants.resourceDelimiter + ErrorDescription.userLoginError);
         return;
       }
 
+      // construct extraQueryParams string from the request
+      let extraQueryParameters = Utils.constructExtraQueryParametersString(request.extraQueryParameters);
+      extraQueryParameters = Utils.addPromptParameter(extraQueryParameters, request.prompt);
+
+      // if the developer provides one of these, give preference to developer choice
+      if (request.account || request.sid || request.loginHint) {
```

Comment:
same comment on factoring out this check

Reply:
Done with the new changes.